### PR TITLE
Support additional address in listeners output

### PR DESCRIPTION
**Pull request by jacob-delgado**
**Please provide a description of this PR:**

Support for dual-stack listeners.

Add `ADDITIONAL ADDRESSES` column and support additional listeners in `--address` flag.

```bash
$ ./out/linux_amd64/istioctl pc listeners sleep-645b966fc4-w2xgk  --address "::"                                                                                                                                            130 ↵
ADDRESS PORT  ADDITIONAL ADDRESSES MATCH                                                                                           DESTINATION
0.0.0.0 80    ::                   Trans: raw_buffer; App: http/1.1,h2c                                                            Route: 80
0.0.0.0 80    ::                   ALL                                                                                             PassthroughCluster
0.0.0.0 8000  ::                   Trans: raw_buffer; App: http/1.1,h2c                                                            Route: 8000
0.0.0.0 8000  ::                   ALL                                                                                             PassthroughCluster
0.0.0.0 15001 ::                   ALL                                                                                             PassthroughCluster
0.0.0.0 15001 ::                   Addr: *:15001                                                                                   Non-HTTP/Non-TCP
0.0.0.0 15006 ::                   Addr: *:15006                                                                                   Non-HTTP/Non-TCP
0.0.0.0 15006 ::                   Trans: raw_buffer; Addr: ::/0                                                                   InboundPassthroughClusterIpv6
0.0.0.0 15006 ::                   Trans: tls; App: istio-http/1.0,istio-http/1.1,istio-h2; Addr: ::/0                             InboundPassthroughClusterIpv6
0.0.0.0 15006 ::                   Trans: raw_buffer; App: http/1.1,h2c; Addr: ::/0                                                InboundPassthroughClusterIpv6
0.0.0.0 15006 ::                   Trans: tls; App: TCP TLS; Addr: ::/0                                                            InboundPassthroughClusterIpv6
0.0.0.0 15006 ::                   Trans: tls; Addr: ::/0                                                                          InboundPassthroughClusterIpv6
0.0.0.0 15006 ::                   Trans: raw_buffer; App: http/1.1,h2c; Addr: 0.0.0.0/0                                           InboundPassthroughClusterIpv4
0.0.0.0 15006 ::                   Trans: tls; App: TCP TLS; Addr: 0.0.0.0/0                                                       InboundPassthroughClusterIpv4
0.0.0.0 15006 ::                   Trans: raw_buffer; Addr: 0.0.0.0/0                                                              InboundPassthroughClusterIpv4
0.0.0.0 15006 ::                   Trans: tls; Addr: 0.0.0.0/0                                                                     InboundPassthroughClusterIpv4
0.0.0.0 15006 ::                   Trans: tls; App: istio-http/1.0,istio-http/1.1,istio-h2; Addr: 0.0.0.0/0                        InboundPassthroughClusterIpv4
0.0.0.0 15006 ::                   Trans: tls; App: istio,istio-peer-exchange,istio-http/1.0,istio-http/1.1,istio-h2; Addr: *:8000 Cluster: inbound|8000||
0.0.0.0 15006 ::                   Trans: raw_buffer; Addr: *:8000                                                                 Cluster: inbound|8000||
0.0.0.0 15010 ::                   Trans: raw_buffer; App: http/1.1,h2c                                                            Route: 15010
0.0.0.0 15010 ::                   ALL                                                                                             PassthroughCluster
0.0.0.0 15014 ::                   Trans: raw_buffer; App: http/1.1,h2c                                                            Route: 15014
0.0.0.0 15014 ::                   ALL                                                                                             PassthroughCluster
```

### CODE REVIEW

File: istioctl/pkg/writer/envoy/configdump/listener.go
Comment (howardjohn) [inline istioctl/pkg/writer/envoy/configdump/listener.go:62]: This change LGTM

File: istioctl/pkg/writer/envoy/configdump/listener.go
Comment (howardjohn) [inline istioctl/pkg/writer/envoy/configdump/listener.go:315]: I don't think we should add this part. Its adding a lot of bloat to the output without providing much value.

if we really need to put all of them, I would just use the same column with CSV:

```
ADDRESS
0.0.0.0,::
```

**Comment by jacob-delgado** [comment]
New output

```shell
./out/linux_amd64/istioctl pc listeners sleep-9454cc476-wxbgw                                                                                                   130 ↵ ✹
ADDRESSES     PORT  MATCH                                                                                         DESTINATION
10.96.0.10    53    ALL                                                                                           Cluster: outbound|53||kube-dns.kube-system.svc.cluster.local
0.0.0.0,::    80    Trans: raw_buffer; App: http/1.1,h2c                                                          Route: 80
0.0.0.0,::    80    ALL                                                                                           PassthroughCluster
10.96.0.1     443   ALL                                                                                           Cluster: outbound|443||kubernetes.default.svc.cluster.local
10.96.19.29   443   ALL                                                                                           Cluster: outbound|443||metrics-server.kube-system.svc.cluster.local
```

File: istioctl/pkg/writer/envoy/configdump/listener.go
Comment (hanxiaop) [inline istioctl/pkg/writer/envoy/configdump/listener.go:338]: The output LGTM. One question: do we need to sort the addresses here in some order? Are ipv6 addresses always in additionalAddresses?

File: istioctl/pkg/writer/envoy/configdump/listener.go
Comment (zhlsunshine) [reply istioctl/pkg/writer/envoy/configdump/listener.go]: > I don't think we should add this part. Its adding a lot of bloat to the output without providing much value.
> 
> if we really need to put all of them, I would just use the same column with CSV:
> 
> ```
> ADDRESS
> 0.0.0.0,::
> ```

Hi @howardjohn & @jacob-delgado, if the address contains `port`, it may look like this:

```
ADDRESS
0.0.0.0:80,[::]:80
```

File: istioctl/pkg/writer/envoy/configdump/listener.go
Comment (zhlsunshine) [reply istioctl/pkg/writer/envoy/configdump/listener.go:338]: I think it's unnecessary to sort the order because the order of the IP addresses may be dominated by many factors, such as the k8s cluster's setting and how the application is deployed.

**Comment by zhlsunshine** [comment]
Hi @jacob-delgado, thanks for this PR, LGTM.

File: releasenotes/notes/istioctl-additional-address.yaml
Comment (hanxiaop) [inline releasenotes/notes/istioctl-additional-address.yaml:8]:
```suggestion
    **Added** support for displaying multiple addresses of listeners in `istioctl proxy-config listeners`.
```

**Review by hanxiaop — APPROVED** [review]
LGTM, after the minor issue in the release notes is addressed. I think it's better to include the command name to help release managers understand the feature better. I've added a DNM label, and feel free to remove it once the comment is addressed.

**Comment by zhlsunshine** [comment]
LGTM

**Pull request merged**
